tests/installation/test_executor.py

```diff
@@ -187,6 +203,64 @@ def test_execute_should_show_errors(config, mocker, io, env):
     assert expected in io.fetch_output()
 
 
+def test_execute_works_with_ansi_output(
+    mocker, config, pool, io_decorated, tmp_dir, mock_file_downloads, env
+):
+    config = Config()
+    config.merge({"cache-dir": tmp_dir})
+
+    executor = Executor(env, pool, config, io_decorated)
+
+    mocker.patch.object(env, "_run", return_value=0)
```

Comment:
Since the issue comes from the fact that it currently returns the standard output, this patch should also return a string an not an integer, otherwise this test does not test the fix we need (as-is it works even on the currently broken code).

Reply:
fixed!